cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/DeleteScheduleStage.java

```diff
@@ -56,10 +56,12 @@ public void process(ApplicationWithPrograms input) throws Exception {
       // delete schedules that existed in the old app spec, but don't anymore
       ScheduleSpecification scheduleSpec = entry.getValue();
       ProgramType programType = ProgramType.valueOfSchedulableType(scheduleSpec.getProgram().getProgramType());
-      LOG.debug("Deleting schedule {} with specification {}", entry.getKey(), scheduleSpec);
-      scheduler.deleteSchedule(input.getApplicationId().program(programType,
-                                                                scheduleSpec.getProgram().getProgramName()),
-                               scheduleSpec.getProgram().getProgramType(), scheduleSpec.getSchedule().getName());
+      if (input.canUpdateSchedules()) {
```

Comment:
Will it be better to just check the flag at the very beginning and return if canUpdateSchedule is set to false like you do in CreateScheduleStage